### PR TITLE
optimize skiplayernorm by tune parameters with thread per block

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu [hunk 1]
```diff
@@ -185,10 +185,22 @@ Status LaunchSkipLayerNormKernel(
...
-      constexpr int block_size = 256;
-      SkipLayerNormKernel<T, block_size, Simplified>
-          <<<grid_size, block_size, 0, stream>>>(ld, input, skip, beta, gamma, bias,
-                                                 maybe2half<T>(epsilon), output, skip_input_bias_add_output);
+      int cuda_device_id{0};
+      int max_threads_per_block{1024};
+      CUDA_CALL_THROW(cudaGetDevice(&cuda_device_id));
+      CUDA_CALL_THROW(cudaDeviceGetAttribute(&max_threads_per_block, cudaDevAttrMaxThreadsPerBlock, cuda_device_id));
+
+      if (max_threads_per_block >= 1024 && ld > (512 * 4)) {
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu:193]: You can pass max_threads_per_block from Attention class.

**Review by yufenglee — APPROVED** [review]

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu:193]: Just keep current api.

Comment (tianleiwu) [reply onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu:193]: The function calls of cudaGetDevice and cudaDeviceGetAttribute takes time. There is no need to call them in each kernel launch if they can be done only once.

**Comment by zhanghuanrong** [comment]
Need total rewrite skiplayernorm for correctness.

**Pull request closed without merging**